diff --git a/drivers/net/wireless/wcnss/wcnss_wlan.c b/drivers/net/wireless/wcnss/wcnss_wlan.c
index 5f3f453..f75c81b 100644
--- a/drivers/net/wireless/wcnss/wcnss_wlan.c
+++ b/drivers/net/wireless/wcnss/wcnss_wlan.c
@@ -1869,6 +1869,13 @@ static void wcnss_nvbin_dnld(void)
 	pr_info("wcnss: NV file name = %s\n", nv_file_name);
 	ret = request_firmware(&nv, nv_file_name, dev);
 
+    pr_err("ret: %d\n", ret);
+    if (!nv)
+        pr_err("nv: null\n");
+    if (!nv->data)
+        pr_err("nv->data: null\n");
+    if (!nv->size)
+        pr_err("nv->size: null\n");
 	if (ret || !nv || !nv->data || !nv->size) {
 		pr_err("wcnss: %s: request_firmware failed for %s\n",
 			__func__, nv_file_name);
-- 
2.7.4

